src/api/transaction/order.js

```diff
@@ -2,40 +2,45 @@
 'use strict';
 const utils = require('./utils');
 const validate = utils.common.validate;
-const Transaction = utils.common.core.Transaction;
+const offerFlags = utils.common.txFlags.OfferCreate;
 import type {Instructions, Prepare} from './types.js';
 import type {Order} from '../ledger/transaction-types.js';
 
-const OfferCreateFlags = {
-  passive: {set: 'Passive'},
-  immediateOrCancel: {set: 'ImmediateOrCancel'},
-  fillOrKill: {set: 'FillOrKill'}
-};
-
-function createOrderTransaction(account: string, order: Order): Transaction {
+function createOrderTransaction(account: string, order: Order): Object {
   validate.address(account);
   validate.order(order);
 
-  const transaction = new Transaction();
   const takerPays = utils.common.toRippledAmount(order.direction === 'buy' ?
     order.quantity : order.totalPrice);
   const takerGets = utils.common.toRippledAmount(order.direction === 'buy' ?
     order.totalPrice : order.quantity);
 
-  transaction.offerCreate(account, takerPays, takerGets);
-
-  utils.setTransactionBitFlags(transaction, order, OfferCreateFlags);
+  const txJSON: Object = {
+    TransactionType: 'OfferCreate',
+    Account: account,
+    TakerGets: takerGets,
+    TakerPays: takerPays,
+    Flags: 0
+  };
   if (order.direction === 'sell') {
-    transaction.setFlags('Sell');
+    txJSON.Flags |= offerFlags.Sell;
```

Comment:
[Here](https://github.com/ripple/ripple-lib/blob/develop/src/api/transaction/utils.js#L53) there is comment that says that we should convert flags to unsigned after doing bitwise operations. Do we need it?